plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlVectorType.java

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    @Test
+    void testVectorWrite()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector_writes", "(v vector(1))")) {
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES ARRAY[REAL '1.0'], ARRAY[REAL '3.4028235e+38f'], ARRAY[REAL '1.4E-45'], NULL", 4);
+            assertThat(query("SELECT * FROM " + table.getName()))
+                    .matches("VALUES ARRAY[REAL '1.0'], ARRAY[REAL '3.4028235e+38f'], ARRAY[REAL '1.4E-45'], NULL");
+
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES ARRAY[REAL '1.0', REAL '2.0']", ".*expected 1 dimensions, not 2.*");
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES ARRAY[nan()]", "(?s).*NaN not allowed in vector.*");
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES ARRAY[-infinity()]", "(?s).*infinite value not allowed in vector.*");
+            assertQueryFails("INSERT INTO " + table.getName() + " VALUES ARRAY[+infinity()]", "(?s).*infinite value not allowed in vector.*");
+        }
+    }
+
+    @Test
+    void testVectorNull()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector_null", "(id int, v vector(1))")) {
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES (1, NULL), (2, '[1]')");
+            assertThat(query("SELECT v FROM " + table.getName()))
+                    .matches("VALUES CAST(NULL AS ARRAY(REAL)), ARRAY[REAL '1.0']");
+            assertThat(query("SELECT id FROM " + table.getName() + " WHERE v IS NULL"))
+                    .matches("VALUES 1");
+            assertThat(query("SELECT id FROM " + table.getName() + " WHERE v IS NOT NULL"))
+                    .matches("VALUES 2");
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES (3, '[NULL]')"))
+                    .hasMessageContaining("ERROR: invalid input syntax for type vector");
+        }
+    }
+
+    @Test
+    void testVectorArbitraryDimension()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector_arbitrary", "(id int, v vector)")) {
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES (1, '[1]'), (2, '[1,2]'), (3, '[1,2,3]')");
+            assertThat(query("SELECT v FROM " + table.getName()))
+                    .matches("VALUES ARRAY[REAL '1.0'], ARRAY[REAL '1.0', REAL '2.0'], ARRAY[REAL '1.0', REAL '2.0', REAL '3.0']");
+
+            onRemoteDatabase().execute("SELECT v <-> '[4,5,6]' FROM " + table.getName() + " WHERE id = 3");
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <-> '[4,5,6]' FROM " + table.getName()))
+                    .hasMessageContaining("different vector dimensions");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES (4, '[]')"))
```

Comment:
What if the insert an empty array ?

Reply:
What do you mean? This line tries inserting an empty array.